group-coordinator/src/main/java/org/apache/kafka/coordinator/group/consumer/ConsumerGroup.java

```diff
@@ -592,6 +607,45 @@ public void validateOffsetFetch(
         validateMemberEpoch(memberEpoch, member.memberEpoch());
     }
 
+    /**
+     * Validates the OffsetDelete request.
+     */
+    @Override
+    public void validateOffsetDelete() throws GroupIdNotFoundException {
+        if (state() == ConsumerGroupState.DEAD) {
+            throw new GroupIdNotFoundException(String.format("Group %s is in dead state.", groupId));
+        }
+    }
+
+    /**
+     * Validates the GroupDelete request.
+     */
+    @Override
+    public void validateGroupDelete() throws ApiException {
+        if (state() == ConsumerGroupState.DEAD) {
+            throw new GroupIdNotFoundException(String.format("Group %s is in dead state.", groupId));
+        } else if (state() == ConsumerGroupState.STABLE
+            || state() == ConsumerGroupState.ASSIGNING
+            || state() == ConsumerGroupState.RECONCILING) {
+            throw Errors.NON_EMPTY_GROUP.exception();
+        }
+
+        // We avoid writing the tombstone when the generationId is 0, since this group is only using
+        // Kafka for offset storage.
+        if (groupEpoch() <= 0) {
+            throw Errors.UNKNOWN_SERVER_ERROR.exception();
+        }
```

Comment:
This does not seem correct to me because this exception does not apply to consumer groups.